components/Layout/Images/LazyImage.vue

```diff
@@ -0,0 +1,114 @@
+<template>
+  <img :data-src="src" :alt="alt" class="lazy-hidden" @error="onError" />
+</template>
+
+<script lang="ts">
+import Vue from 'vue';
+import { sanitize } from 'dompurify';
+
+/**
+ * Swaps the 'data-src' attribute with 'src' attribute.
+ *
+ * @param {HTMLElement} elem - DOM node of the target img element
+ * @param {string} url - url of the image
+ *
+ */
+function fillImageElement(elem: HTMLElement, url: string): void {
+  const preloaderImg = new Image();
+  preloaderImg.src = url;
+
+  preloaderImg.addEventListener('load', () => {
+    requestAnimationFrame(() => {
+      elem.setAttribute('src', url);
+      elem.removeAttribute('data-src');
+      elem.classList.remove('lazy-hidden');
+      elem.classList.add('lazy-show');
+    });
+  });
+}
```

Comment:
There needs to be a clean up function to remove the event listener

Reply:
The event listener is bound to an offscreen element [here](https://github.com/jellyfin/jellyfin-vue/pull/408/files#diff-4d53a82f6d48248fa1a22bd2f449b0604526d455c84d8b528ab723182b8f4c76R17) and, as far as I know, should be garbage collected as soon the code exits from the function's context.

I can add a ``delete preloaderImg`` if we want to be 100% sure though.